dot/sync/chain_sync.go

```diff
@@ -169,23 +176,48 @@ func newChainSync(cfg chainSyncConfig) *chainSync {
 	}
 }
 
+func (cs *chainSync) waitEnoughPeersAndTarget() <-chan struct{} {
+	resultCh := make(chan struct{})
+	go func() {
+		defer close(resultCh)
+		for {
+			select {
+			case <-resultCh:
+				return
+			default:
+			}
```

Comment:
why do we read from `resultCh` here? Who writes to this channel?

Reply:
Answered here https://github.com/ChainSafe/gossamer/pull/3392#issuecomment-1788953984